src/components/weather/WeatherByCity.tsx

```diff
@@ -0,0 +1,39 @@
+import React  from 'react'
+import { RemoveButton, SavedCities, SavedCity } from './styled';
+import { WeatherData } from '../../services/weatherService';
+
+interface WeatherByCityProps {
+    savedCities: WeatherData[];
+    setSavedCities: React.Dispatch<React.SetStateAction<WeatherData[]>>;
+}
+  
+const WeatherByCity: React.FC<WeatherByCityProps> = ({ savedCities, setSavedCities }) => {
```

Comment:
this is wrong, component is called `WeatherByCity` and you are passing list of saved cities.

You should pass only one city.

in WeatherPage.tsx you should adapt component that you map throught all cities and pass only one city
``` 
{savedCities.map((city, index) => (
        <WeatherByCity city={city} setSavedCities={setSavedCities} />
))
}
```

Also props in `WeatherByCity` should called `city` not `savedCities`